cmd/kube-apiserver/app/server.go

```diff
@@ -659,6 +660,24 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
 			return options, fmt.Errorf("failed to build token generator: %v", err)
 		}
 		s.ServiceAccountTokenMaxExpiration = s.Authentication.ServiceAccounts.MaxExpiration
+
+		var pubKeys []interface{}
+		for _, f := range s.Authentication.ServiceAccounts.KeyFiles {
+			keys, err := keyutil.PublicKeysFromFile(f)
+			if err != nil {
+				return options, fmt.Errorf("failed to parse service-account-key-file %q: %v", f, err)
+			}
+			pubKeys = append(pubKeys, keys...)
+		}
+		// TODO(mtaufen): Ensure this is feature-gated
```

Comment:
I assume we'll have to progress this feature through alpha/beta/ga stages.

TODO: also feature gate the extension to system:public-info-viewer

Reply:
That said, the tests treat this as part of the TokenRequestProjection feature, which is currently beta, so perhaps that's another route? @mikedanese @liggitt wdyt?